design-proposals/volume-migration/volume-migration.md

```diff
@@ -0,0 +1,466 @@
+
+Certain types of volumes and disks aren't suitable to be migrated or the feature isn't supported yet. Those migrated volumes will be marked as rejected. 
+If there are other valid volumes belonging to the same VMI as the rejected ones, the VM migration won't be created for those volumes as well. Those volumes will be listed as `Valid` in the `VolumeMigrationStates` field.
+
+The current types of rejected volumes are:
+  * Hotpluggable volumes, this case will be considered in a later iteration.
+  * Filesystem volumes, currently virtiofs doesn't support live-migration.
+  * Shareable volumes, if a disk is shared between multiple VMs, it cannot be safely copied.
+  * LUNs disks, originally the disk might support SCSI protocol but the destination PVC class does not. This case will be considered in a later iteration.
+
+The `VolumeMigrationStatus` reports the phase of the migration:
+  * `Pending` if there are no VMI associated to the migrated volumes
+  * `Scheduling` if the VMI migration object has been created but the migration hasn't started yet
+  * `Running` once the VMI migration starts
+  * `Succeeded` if the VMI migration has succeeded
+  * `Failed` if the VMI migration has failed or there are rejected volumes
+
+## Design details and motivation
+
+### Motivation on the approaches
+
+From libvirt there a multiple ways to achieve the storage migration. It is possible to simply copy the storage on the new destination and then swap the original disk with the new one. This approach is the most efficient as the overhead is caused only by the storage copy.
+
+A different approach is to simultaneously move the storage and live-migrate the guest. When the destination storage is not available on the host where the VM is executing, this solution is required. For this case, costs are greater since there is storage as well as live VM migration overhead.
+
+As KubeVirt is designed today and due to Kubernetes design principles of volume pod immutability, it isn't possible to hotplug and unplug new volumes while the pod is still running. KubeVirt already supports volume hotplug by attaching a new volume to a dummy pod and then bind mounting the storage in the `virt-launcher` pod. Sadly, this method fails to address the issue of removing the old PVCs that were expressly declared in the virt-launcher pod.
+Declaring all volumes hotpluggable could be a potential remedy, but we prefer to avoid it because it has been shown to be fragile for some storage providers and opaque to Kubernetes.
+
+The only workable and generic solution for this restriction appears to be to live migrate the VM with storage. Live migration involves the creation of a new virt-launcher pod by KubeVirt that has the destination PVCs attached. The source virt-launcher pod and source PVCs are removed after the migration is finished because they are no longer needed.
```

Comment:
> The source virt-launcher pod and source PVCs are removed after the migration is finished because they are no longer needed.

This might be a bit confusing. According to the design, the PVCs are handled respecting the specified reclaim policy (either delete or retain).

Reply:
You're right. The sentence also doesn't explain much, I'd simply remove it